### ISSUE TITLE
Android Text lineHeight is incorrect for the first line

### ISSUE THREAD

**Issue by brentvatne** [facebook/react-native#7546]
Noticed this in a production app, easy repro case with the template app: all I did to the example app was add some `lineHeight` and change the text to make it obvious. You can copy and paste the source from here: https://gist.github.com/brentvatne/ed98472bc8605484862c7adfdee8d5ca

#### Example gif: 
![wrong](https://cloud.githubusercontent.com/assets/90494/15238792/26d983c2-1891-11e6-93c3-6929ff57f9f8.gif)

Full video: http://url.brentvatne.ca/1jQXm

cc @dmmiller

**Comment by dmmiller** [comment]
@andreicoman11 any thoughts?

**Comment by andreicoman11** [comment]
Will have to investigate, it's probably a cool bug somewhere.

**Comment by aaronschubert0** [comment]
@brentvatne @andreicoman11 Did this ever get fixed?

**Comment by andreicoman11** [comment]
I have diff internally that fixes this, I just need to get all details right and land it.

**Comment by aaronschubert0** [comment]
@andreicoman11 That'd be awesome :D

**Comment by brentvatne** [comment]
wooo awesome @andreicoman11!

**Comment by alexHlebnikov** [comment]
Same problem here, is there any fix for this?

**Comment by alexHlebnikov** [comment]
@andreicoman11 is there any chance you will do a PR with a fix?

**Comment by andreicoman11** [comment]
https://github.com/facebook/react-native/commit/c79f6177426a417299b5eb13d483613c97b303b3 fixes this.

The TLDR is that Android's StaticLayout gives you incorrect layout when using `lineHeight`. In a simple test app, this is what it looked like: 
![screenshot-2016-08-22_12 23 05 172](https://cloud.githubusercontent.com/assets/6615092/18436950/bdd033f4-78f2-11e6-834b-0d0a573c217e.png)

It took longer than usual to fix this because it was tricky to get the implementation details right. We initially pushed a version of this fix internally and then had to immediately revert it because it didn't work right in many cases. Given that this might change the layout of many Text components, we might still need to do that, or to add more fixes on top of this one.

**Comment by andreicoman11** [comment]
@alexHlebnikov do you mind opening a new issue with the entire code, expected outcome, actual outcome, maybe also a few screenshots? Thanks!

**Comment by jreziga** [comment]
Please try to use .TTF webfont. I had this issue but I was using .OTF

**Comment by mdvacca** [comment]
Same problem here with OTF or TTF fonts. (the lineHeight is incorrect only for the first line)

I'm using
RN 0.34.RC.0
Nexus 5 (Android 5.0.0 and Android 6.0.0)

**Comment by mdvacca** [comment]
I've created a custom Text component that handles lineHeight correctly in Android, while waiting for this to be fixed.

https://gist.github.com/mdvacca/934ae7b9318c69d11f3965093641a64d#file-text-js
